### ISSUE TITLE
Support per-workspace disk space quota

### ISSUE THREAD

**Issue by csweichel** [gitpod-io/gitpod#4938]
Workspaces don't have an enforced disk space quota right now. We have had two attempts at implementing this in the past:
1. mounting a disk image using the loopback device: was rather cumbersome and had difficult to manage failure modes that lead to data loss
2. ext4 quota: are not enforceable, but only tally up the used disk space

One possible way forward would be to add XFS quota support to ws-daemon. If the workspace volume is an XFS volume, we would set a configurable quota on that workspace directory.

**Comment by csweichel** [comment]
Related KEP 1029: https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/1029-ephemeral-storage-quotas/README.md

**Comment by csweichel** [comment]
We need to move out of GKE before we can actually operate or develop this change. Will re-add once we've made progress on that end.

**Comment by wulfthimm** [comment]
Is this still an issue as we are now using a raid0 with XFS?

**Comment by aledbf** [comment]
@csweichel can we prioritize this change? A side effect of not having quotas is that when we create backups, is the error `maximum size exceeded` and the memory spike after that task.

**Comment by aledbf** [comment]
Yesterday we had a full disk event where one workspace was using almost 1TB.

**Comment by meysholdt** [comment]
Post-Mortem issue from yesterday's incident: https://github.com/gitpod-io/security/issues/4 I'm closing it as a duplicate of this issue.

**Comment by roboquat** [comment]
@csweichel: Issue scheduled in the workspace team (WIP: 0)

<details>

In response to [this](https://github.com/gitpod-io/gitpod/issues/4938#issuecomment-925910776):

>/schedule


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by csweichel** [comment]
requires https://github.com/gitpod-io/gitpod/issues/5826
